evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -133,22 +136,23 @@ def transform(self, X, y=None):
         Returns:
             pd.DataFrame: Transformed data, where each categorical feature has been encoded into numerical columns using one-hot encoding.
         """
-        X_ww = infer_feature_types(X)
-        X_copy = self._handle_parameter_handle_missing(X_ww)
+        X = infer_feature_types(X)
+        original_ltypes = X.ww.schema.logical_types
+        X_copy = self._handle_parameter_handle_missing(X)
 
-        X_ww = X_ww.ww.drop(self.features_to_encode)
+        X = X.drop(columns=self.features_to_encode)
```

Comment:
Cool, this is an example of where we don't need to use WW right?

Reply:
Yes

Reply:
Ah, so the woodwork schema does end up getting updated in this case?

Reply:
We retain as much as we can when with `retain_custom_types...` but now that I think about this, I edited this when I was debugging our memory issues. Since we're not doing the drop in-place, I think we can do `X.ww.drop`!